### PR TITLE
Reimplement coordshort request handler.

### PR DESCRIPTION
**Pull request by mpoeter**
The old implementation sent `statistcs/short` requests to all coordinators (including itself). The `statistics/short` handler is also implemented in JS and runs two DB queries. So for a cluster with 3 coordinators, the old implementation issued 3 network requests, which triggered a total of 6 DB queries and required 4 V8 contexts (one for the initial coordshort request, and one for each statistic/short request).

The new implementation only runs two DB queries without any additional requests to other coordinators, resulting in reduced load on the cluster.

In addition this reimplementation fixes the way statistics samples of coordinators are merged. Previously we simply took the timestamps from the samples of one coordinator and simply merged all samples based on their index. However, coordinators don't necessarily write their statistics sample at the same time, and it is also not guaranteed that all coordinators have the same number of samples.

### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.6, 3.7

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** in shell_server
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12917/

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 1]
```diff
@@ -406,152 +407,57 @@ router.use((req, res, next) => {
...
+      FOR s IN _statistics15
+        FILTER s.time > @start
+        FILTER s.clusterId IN @clusterIds
+        SORT s.time
+        LET clientConnectionsCurrent = s.client.httpConnections
+        LET time = s.time
+        COLLECT clusterId = s.clusterId INTO server KEEP clientConnectionsCurrent, time
+        LET clientConnectionsCurrent = LAST(server).clientConnectionsCurrent
+        COLLECT AGGREGATE clientConnections15M = SUM(clientConnectionsCurrent)
+        RETURN {clientConnections15M: clientConnections15M || 0}`;
```
Comment (jsteemann) [inline js/apps/system/_admin/aardvark/APP/statistics.js:424]:
```suggestion
        COLLECT clusterId = s.clusterId INTO serverConnections = s.client.httpConnections
        LET clientConnectionsCurrent = LAST(serverConnections)
        COLLECT AGGREGATE clientConnections15M = SUM(clientConnectionsCurrent)
        RETURN {clientConnections15M: clientConnections15M || 0}`;
```

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 2]
```diff
@@ -406,152 +407,57 @@ router.use((req, res, next) => {
...
+    const statsSampleQuery = `
+      FOR s IN _statistics
+        FILTER s.time > @start
+        FILTER s.clusterId IN @clusterIds
+        RETURN {
+          time: s.time,
+          clusterId: s.clusterId,
+          avgRequestTime: s.client.avgRequestTime,
+          bytesSentPerSecond: s.client.bytesSentPerSecond,
+          bytesReceivedPerSecond: s.client.bytesReceivedPerSecond,
```
Comment (jsteemann) [inline js/apps/system/_admin/aardvark/APP/statistics.js:435]: IMHO this should also return `physicalMemory`, `residentSize` etc., as these are accessed by `MergeStatistics(...)`.

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 3]
```diff
@@ -406,152 +407,57 @@ router.use((req, res, next) => {
...
-  if (coordinatorStats) {
-    res.json({'enabled': coordinatorStats.some(stat => stat.enabled), 'data': merged});
-  } else {
-    res.json({
-      'enabled': false,
-      'data': {}
-    });
-  }
+
+  res.json({'enabled': Object.keys(merged).length !== 0 , 'data': merged});
```
Comment (jsteemann) [inline js/apps/system/_admin/aardvark/APP/statistics.js:460]: `enabled` should be determined differently.

**Review by jsteemann — CHANGES_REQUESTED** [review]

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 4]
```diff
@@ -406,152 +407,59 @@ router.use((req, res, next) => {
...
+  var merged = { };
+
+  const coordinators = global.ArangoClusterInfo.getCoordinators();
+  try {
+    const stats15Query = `
+      FOR s IN _statistics15
+        FILTER s.time > @start
+        FILTER s.clusterId IN @clusterIds
+        SORT s.time
+        LET clientConnectionsCurrent = s.client.httpConnections
```
Comment (jsteemann) [inline js/apps/system/_admin/aardvark/APP/statistics.js:419]:
```suggestion
```

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12925/

**Review by jsteemann — APPROVED** [review]

**Pull request merged**